torch/_inductor/codegen/cuda/cutlass_python_evt.py

```diff
@@ -0,0 +1,240 @@
+            codegen.get_writes(),
+            codegen.get_renames(),
+            codegen.get_value(),
+        )
+
+    def __init__(self, accumulator_node_name: str, last_usages: OrderedSet[str]):
+        """
+
+        Initializes a CutlassEVTEpilogueArgumentFormatter object. Do not instantiate directly.
+        Use the CutlassEVTCodegen.ir_to_evt_python_code static method.
+
+        Args:
+            accumulator_node_name: The name of the accumulator node which should contain
+                                          the Matmul result before fusion according to the IR graph.
+            epilogue_nodes: The list of scheduler nodes to be fused into the epilogue
+        """
+        self.accumulator_node_name: str = accumulator_node_name  #
+        self.output: IndentedBuffer = IndentedBuffer(1)  # The output buffer for codegen
+        self.var_counter: Iterator[int] = itertools.count()
+        self.store_name_to_value: dict[str, OpsValue] = (
+            dict()
+        )  # Aliases for subexpression functors
+        self.reads: OrderedSet[str] = OrderedSet()
+        self.last_usages: OrderedSet[str] = OrderedSet()
+        self.cur_node: Optional[ComputedBuffer] = None
+
+        if accumulator_node_name not in last_usages:
+            self.store(accumulator_node_name, value=OpsValue(_ACCUMULATOR_ALIAS))
+
+    def get_value(self) -> str:
+        return linesep.join(
+            [
+                self._render_input_signature(),
+                self.output.getvalue(),
+                self._render_return_statement(),
+            ]
+        )
+
+    @contextmanager
+    def set_cur_node(self, node: ComputedBuffer) -> Generator[None, Any, Any]:
+        prev_node = self.cur_node
+        try:
+            self.cur_node = node
+            yield
+        finally:
+            self.cur_node = prev_node
+
+    def get_renames(self) -> dict[str, str]:
+        renames = {k: v.value for k, v in self.store_name_to_value.items()}
+        renames[self.accumulator_node_name] = _ACCUMULATOR_ALIAS
+        return renames
+
+    def get_reads(self) -> list[str]:
+        return list(self.reads)
+
+    def get_writes(self) -> list[str]:
+        return list(self.store_name_to_value.keys())
+
+    def load(self, name: str, index: Any) -> str:
+        self._check_indexing(name, index)
+        if name == self.accumulator_node_name:
+            self.reads.add(name)
+            return _ACCUMULATOR_ALIAS
+        elif name in self.store_name_to_value:
+            return self.store_name_to_value[name].value
+        else:
+            self.reads.add(name)
+            return name
+
+    def store(
+        self, name: Any, index: Any = None, value: Any = None, mode: Any = None
+    ) -> None:
+        if name not in self.last_usages:
+            value_to_write = value
+            if not self.store_name_to_value:
+                # EVT requires an output to be named D lol
+                # so rename the first store to D
```

Comment:
Do we need to check `_check_indexing` here as well ?

Reply:
I don't think it's possible to have different indexing on stores because ComputedBuffers always grab the indexer from its layout, there isn't a way to intercept the indexing to change it like would happen with a PermuteView. TBH I'll add it just to be safe.